pkg/client/unversioned/clientcmd/overrides.go

```diff
@@ -105,6 +108,20 @@ func (f FlagInfo) BindBoolFlag(flags *pflag.FlagSet, target *bool) {
 	}
 }
 
+// BindDurationFlag binds the flag based on the provided info.  If LongName == "", nothing is registered
+func (f FlagInfo) BindDurationFlag(flags *pflag.FlagSet, target *time.Duration) {
```

Comment:
I still believe `--request-timeout` should support values without units, which defaults to seconds like every other timeout flag out there. Like in `ssh`, `timeout`, `scp`, `curl`, and so on. 

You'd probably have to use string and parse it by yourself like in [here](https://play.golang.org/p/LgfIAgEvDC).